AirCasting/CoreData/MeasurementStreamStorage.swift

```diff
@@ -67,6 +91,29 @@ final class HiddenCoreDataMeasurementStreamStorage: MeasurementStreamStorageCont
         self.context = context
     }
     
+    func deleteSession(_ sessionUUID: SessionUUID) throws {
+        do {
+            try context.delete(context.existingSession(uuid: sessionUUID))
+        } catch {
+            Log.info("## Error when deleting session")
+        }
+    }
+    
+    func deleteStreams(_ sessionUUID: SessionUUID) throws {
+        let sessionEntity = try context.existingSession(uuid: sessionUUID)
```

Comment:
I have mixed feelings about this. In general when dealing with outside client code we really want to get some sort of `id` instead of a `NSManagedObject` instance. Fetching it should be really fast and also delete is not that time sensitive.
The reason is that we are always wrapping this code inside the `context.performBlock { .. }` for this class and if someone passes us a `Session` fetched from different context the app will most likely crash or do some weird stuff

Reply:
Ok, so maybe we could rename this property? tbh I didn't notice it's just UUID until now, because its name is sessionEntity :D